src/charm.py

```diff
@@ -6,56 +6,43 @@
 
 import logging
 
-from charms.storage_libs.v0.nfs_interfaces import NFSProvides, ShareRequestedEvent
-from ops.charm import CharmBase
-from ops.framework import StoredState
-from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus
+import ops
+from charms.filesystem_client.v0.filesystem_info import FilesystemProvides, NfsInfo
 
 logger = logging.getLogger(__name__)
 
 
-class NFSServerProxyCharm(CharmBase):
+class NFSServerProxyCharm(ops.CharmBase):
     """NFS server proxy charmed operator."""
 
-    _stored = StoredState()
-
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
-        self._stored.set_default(endpoint=None)
-        self._nfs_share = NFSProvides(self, "nfs-share")
+        self._filesystem = FilesystemProvides(self, "filesystem", "server-peers")
         self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self._nfs_share.on.share_requested, self._on_share_requested)
 
     def _on_config_changed(self, _) -> None:
         """Handle updates to NFS server proxy configuration."""
-        if (endpoint := self.config.get("endpoint")) is None:
-            self.unit.status = BlockedStatus("No configured endpoint")
+        if (hostname := self.config.get("hostname")) is None:
+            self.unit.status = ops.BlockedStatus("No configured hostname.")
+            return
+
+        if (path := self.config.get("path")) is None:
+            self.unit.status = ops.BlockedStatus("No configured path.")
```

Comment:
[nit]: See my comment above ^

```suggestion
            self.unit.status = ops.BlockedStatus("No configured path")
```